### PR TITLE
gitAndTools: move everything to the top level

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

The comment at the top of git-and-tools/default.nix said:

    /* All git-relates tools live here, in a separate attribute set so that users
     * can get a fast overview over what's available.

but unfortunately that hasn't actually held up in practice.

Git-related packages have continued to be added to the top level, or
into gitAndTools, or sometimes both, basically at random, so having
gitAndTools is just confusing.  In fact, until I looked as part of
working on getting rid of gitAndTools, one program (ydiff) was
packaged twice independently, once in gitAndTools and once at the top
level (I fixed this in 98c34901969).

So I think it's for the best if we move away from gitAndTools, and
just put all the packages it previously contained at the top level.
I've implemented this here by just making gitAndTools an alias for the
top level -- this saves having loads of lines in aliases.nix.  This
means that people can keep referring to gitAndTools in their
configuration, but it won't be allowed to be used within Nixpkgs, and
it won't be presented to new users by e.g. nix search.

The only other change here that I'm aware of is that
appendToName "minimal" is not longer called on the default git
package, because doing that would have necessitated having a private
gitBase variable like before.  I think it makes more sense not to do
that anyway, and reserve the "minimal" suffix only for gitMinimal.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mic92 — APPROVED** [review]

**Review by nicknovitski — APPROVED** [review]
I have no objections to this.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -194,7 +194,15 @@ mapAliases ({
   gdb-multitarget = gdb; # added 2017-11-13
   gdk_pixbuf = gdk-pixbuf; # added 2019-05-22
   gettextWithExpat = gettext; # 2016-02-19
-  git-hub = gitAndTools.git-hub; # added 2016-04-29
+  gitAndTools = self // { # added 2021-01-14
```
Comment (cole-h) [inline pkgs/top-level/aliases.nix:197]: Should we also warn here that `gitAndTools` is deprecated, so people can begin migrating away?

```suggestion
  gitAndTools = builtins.trace "gitAndTools has been deprecated and is now an alias to the top level" (self // { # added 2021-01-14
```

or something.

Comment (alyssais) [reply pkgs/top-level/aliases.nix:197]: I thought about that, but no other aliases have warnings, and I’m not sure it really matters if people are still using gitAndTools.  I was also worried about interactions with imperative package management — would those users have to reinstall all their packages?

But I’m happy to reconsider.  What do you think?

Comment (cole-h) [reply pkgs/top-level/aliases.nix:197]: No, you're right. It being an alias means it's not possible to use it inside Nixpkgs, but what does it really matter if people are using it in their own personal configs?

**Review by cole-h — APPROVED** [review]
Diff LGTM.

**Comment by ryantm** [comment]
Thank you for doing this. Probably going to win my favorite PR of year award ;-)

**Comment by worldofpeace** [comment]
I'm also pleasantly surprised by this, thanks @alyssais

**Pull request merged**